engine/include/graphics/vkHelpers.hpp

```diff
@@ -33,7 +33,9 @@ namespace vk::util
 
 	bool checkDeviceExtensionSupport(vk::PhysicalDevice device, const std::vector<const char *> & deviceExtensions);
 
-	vk::UniqueShaderModule createShaderModule(vk::Device device, vk::ShaderStageFlagBits shaderStage, std::string const & shaderText);
+	vk::UniqueShaderModule createShaderModule(vk::Device device, std::vector<char> const & code);
```

Comment:
`std::span`, and we shouldn't be aliasing `char` for binary data, use `std::byte` (requires some `reinterpret_cast`s at IO boundaries, but is much more strongly typed everywhere else).